### ISSUE TITLE
Failed to map with error:[target0 not in samples]]

### ISSUE THREAD

**Issue by mc261670164** [PaddlePaddle/PaddleDetection#1360]
Ubuntu机器上在训练PaddleDetection中的ppyolo时，我用的VOC格式的数据集，一直卡在一条log这不往下走，log的内容是“2020-09-04 15:33:29,268-WARNING: recv endsignal from outq with errmsg[consumer[consumer-310-1] failed to map with error:[target0 not in samples]]”。如何才能解决这个问题？我的yml如下:
architecture: YOLOv3
use_gpu: false
max_iters: 15000 #20000
log_smooth_window: 20
log_iter: 10
save_dir: save_models
snapshot_iter: 200
metric: VOC
pretrain_weights: https://paddle-imagenet-models-name.bj.bcebos.com/ResNet50_vd_ssld_pretrained.tar
weights: save_models/model_final
num_classes: 1
use_fine_grained_loss: false
use_ema: true
ema_decay: 0.9998

YOLOv3:
  backbone: ResNet
  yolo_head: YOLOv3Head
  use_fine_grained_loss: true

ResNet:
  norm_type: sync_bn
  freeze_at: 0
  freeze_norm: false
  norm_decay: 0.
  depth: 50
  feature_maps: [3, 4, 5]
  variant: d
  dcn_v2_stages: [5]

YOLOv3Head:
  anchor_masks: [[6, 7, 8], [3, 4, 5], [0, 1, 2]]
  anchors: [[10, 13], [16, 30], [33, 23],
            [30, 61], [62, 45], [59, 119],
            [116, 90], [156, 198], [373, 326]]
  norm_decay: 0.
  coord_conv: true
  iou_aware: true
  iou_aware_factor: 0.4
  scale_x_y: 1.05
  spp: true
  yolo_loss: YOLOv3Loss
  nms: MatrixNMS
  drop_block: true

YOLOv3Loss:
  batch_size: 24
  ignore_thresh: 0.7
  scale_x_y: 1.05
  label_smooth: false
  use_fine_grained_loss: true
  iou_loss: IouLoss
  iou_aware_loss: IouAwareLoss

IouLoss:
  loss_weight: 2.5
  max_height: 608
  max_width: 608

IouAwareLoss:
  loss_weight: 1.0
  max_height: 608
  max_width: 608

MatrixNMS:
    background_label: -1
    keep_top_k: 100
    normalized: false
    score_threshold: 0.01
    post_threshold: 0.01

LearningRate:
  base_lr: 0.001
  schedulers:
  - !PiecewiseDecay
    gamma: 0.1
    milestones:
    - 150000
    - 200000
  - !LinearWarmup
    start_factor: 0.
    steps: 4000

OptimizerBuilder:
  optimizer:
    momentum: 0.9
    type: Momentum
  regularizer:
    factor: 0.0005
    type: L2

TrainReader:
  inputs_def:
    fields: ['image', 'gt_bbox', 'gt_class', 'gt_score']
    num_max_boxes: 50
  dataset:
    !VOCDataSet
    anno_path: train.txt
    dataset_dir: dataset/cartoon_face
    use_default_label: false
    with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
      with_mixup: True
    - !MixupImage
      alpha: 1.5
      beta: 1.5
    - !ColorDistort {}
    - !RandomExpand
      fill_value: [123.675, 116.28, 103.53]
    - !RandomCrop {}
    - !RandomFlipImage
      is_normalized: false
    - !NormalizeBox {}
    - !PadBox
      num_max_boxes: 50
    - !BboxXYXY2XYWH {}
  batch_transforms:
  - !RandomShape
    sizes: [320, 352, 384, 416, 448, 480, 512, 544, 576, 608]
    random_inter: True
  - !NormalizeImage
    mean: [0.485, 0.456, 0.406]
    std: [0.229, 0.224, 0.225]
    is_scale: True
    is_channel_first: false
  - !Permute
    to_bgr: false
    channel_first: True
  - !Gt2YoloTarget
    anchor_masks: [[6, 7, 8], [3, 4, 5], [0, 1, 2]]
    anchors: [[10, 13], [16, 30], [33, 23],
              [30, 61], [62, 45], [59, 119],
              [116, 90], [156, 198], [373, 326]]
    downsample_ratios: [32, 16, 8]
  batch_size: 4
  shuffle: true
  mixup_epoch: 25000
  drop_last: true
  worker_num: 8
  bufsize: 4
  use_process: true

EvalReader:
  inputs_def:
    fields:  ['image', 'im_size', 'im_id', 'gt_bbox', 'gt_class', 'is_difficult']
    num_max_boxes: 50
  dataset:
    !VOCDataSet
      anno_path: test.txt
      dataset_dir: dataset/cartoon_face
      use_default_label: false
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
    - !ResizeImage
      target_size: 608
      interp: 2
    - !NormalizeImage
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]
      is_scale: True
      is_channel_first: false
    - !PadBox
      num_max_boxes: 50
    - !Permute
      to_bgr: false
      channel_first: True
  batch_size: 4
  drop_empty: false
  worker_num: 8
  bufsize: 4

TestReader:
  inputs_def:
    image_shape: [3, 608, 608]
    fields: ['image', 'im_size', 'im_id']
  dataset:
    !ImageFolder
      anno_path: dataset/cartoon_face/label_list.txt
      with_background: false
  sample_transforms:
    - !DecodeImage
      to_rgb: True
    - !ResizeImage
      target_size: 608
      interp: 2
    - !NormalizeImage
      mean: [0.485, 0.456, 0.406]
      std: [0.229, 0.224, 0.225]
      is_scale: True
      is_channel_first: false
    - !Permute
      to_bgr: false
      channel_first: True
  batch_size: 1

**Comment by heavengate** [comment]
ppyolo训练需要设置use_fine_grained_loss为true，看你的配置文件里设置的false
```
weights: save_models/model_final
num_classes: 1
use_fine_grained_loss: false
use_ema: true
ema_decay: 0.9998
```

**Comment by mc261670164** [comment]
您好，请问有关于写yml配置文件的教程么？

**Comment by qingqing01** [comment]
@mc261670164 

YML配置参数说明可以参考 https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.4/docs/advanced_tutorials/config_doc/RCNN_PARAMS_DOC.md 。最近会更新下，加入YOLOv3的配置。

YML配置的设计和简要说明参考： https://github.com/PaddlePaddle/PaddleDetection/blob/release/0.4/docs/advanced_tutorials/config_doc/CONFIG_cn.md


一般来说分为:

- 基础配置:  
    可以参考每类算法的相关YML
- 模型配置
    - 围绕算法的[architectures](https://github.com/PaddlePaddle/PaddleDetection/tree/release/0.4/ppdet/modeling/architectures)进行配置，各项参数是architectures的__init__构造函数中的参数。
    - 以及对architectures的子模块进行配置，如backbone、head等。
- 学习率和优化器设置
- 数据配置
   - 训练数据+预处理配置
   - 评估数据+预处理配置
   - 预测+预处理配置
